lxd/device/nic_bridged.go

```diff
@@ -113,6 +113,16 @@ func (d *nicBridged) validateConfig(instConf instance.ConfigReader) error {
 			if dhcpv4Subnet != nil && !dhcpalloc.DHCPValidIP(dhcpv4Subnet, nil, net.ParseIP(d.config["ipv4.address"])) {
 				return fmt.Errorf("Device IP address %q not within network %q subnet", d.config["ipv4.address"], n.Name())
 			}
+
+			ip, _, err := net.ParseCIDR(netConfig["ipv4.address"])
+			if err != nil {
+				return errors.Wrap(err, "unable to parse parent network address")
+			}
+
+			// IP should not be the same as the parent managed network address.
+			if ip.Equal(net.ParseIP(d.config["ipv4.address"])) {
```

Comment:
What happens if the device doesn't have an `ipv4.address` set?

Reply:
It's not captured in the diff, but this whole block is inside a check for just that.